### ISSUE TITLE
twisted.web.guard's funky definition of __all__ thwarts pydoctor's attempt to document it

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3904]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#3904|
|Type|enhancement|
|Created|2009-07-07 08:38:40Z|

If you [look at the API documentation for twisted.web.guard](http://twistedmatrix.com/documents/8.2.0/api/twisted.web.guard.html), you will see that it apparently contains no names at all.

This could be fixed by simply putting constant strings into `__all__` like other modules do.

Attachments:

* [guard-all.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c8b/c8b00df81cbe0798268f3a02cc187d8684cefe3b/5428ef959642cfb94c81e77d6e70bb678a6c6ad0.patch) (474 bytes) - added by exarkun on 2009-07-07 12:16:27Z - change twisted.web.guard's `__all__` to be normal

<details><summary>Searchable metadata</summary>

```
trac-id__3904 3904
type__enhancement enhancement
reporter__glyph glyph
priority__high high
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation__easy documentation, easy
time__1246955920000000 1246955920000000
changetime__1247405846000000 1247405846000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph set owner to @exarkun|
|-|-|

Okay!  Pydoctor generates docs now, land.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set status to `closed`|
|-|-|

(In [27118]) Apply guard-all.patch

Author: exarkun
Reviewer: glyph
Fixes: [#3904](https://github.com/twisted/twisted/issues/3904)

Change the definition of `twisted.web.guard.__all__` so that pydoctor can
interpret it.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
